### PR TITLE
Migrate robotics orin repo to this repo 622

### LINKED ISSUE #622: Migrate robotics-orin repo to this repo
The old code was nuked, the code that currently runs on the jetson is in a different [repository](https://github.com/space-concordia-robotics/robotics-orin)

### PR DESCRIPTION
**Pull request by OZiad**
# Assignee Section

## Description

* Replaced old code with the new stuff from robotics-orin. Up to date with code on master branch in robotics-orin as of May 25th 2024)
* Stuck to the old directory structure for now. (TODO: move gps cpp files to separate directory) 

### Steps for Testing
[Imagine you are the reviewer and did not write the code. Write all the steps necessary to test the feature(s) in an **ordered list**. Be **specific** and include important details here so testers don't need to jump through hoops to test your pull request. Be explicit in all the steps, and make sure the expected output is clearly stated.

1. Launch jetson
2. Go to rover/rospackages
3. build packages
4. Test the following on the rover: a.  wheels functionality(wheels_controller) b. arm control (arm_controller) c. arm ik(arm_ik) 
5. Verify that the behaviour remains unchanged compared to robotics-orin main branch
6. Use ros2 topic echo to monitor logs

closes #622

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [ ] Local Test Performed Successfully

For Pull Requests that do not include code changes, it is not required to perform the tests above.

### CODE REVIEW

File: robot/util/COLCON_IGNORE
Comment (vexandmore) [inline robot/util/COLCON_IGNORE:1]: From my understanding, the purpose of COLCON_IGNORE is to make colcon ignore the containing folder. But this folder is empty?

**Comment by vexandmore** [comment]
Once ik-3 and arcuo-21 are merged into this, I'm ready to approve

File: robot/util/COLCON_IGNORE
Comment (OZiad) [reply robot/util/COLCON_IGNORE:1]: The folder contains utils.py and the rosRoverStart directory

Comment (vexandmore) [reply robot/util/COLCON_IGNORE:1]: I see, looks like those files aren't showing up in the diff view

**Pull request merged**